active_learning/active_learning.py

```diff
@@ -44,35 +52,50 @@ class MockOracle(Oracle):
     MockOracle rates all meta-paths with 1 and never wants to continue to rate more.
     """
 
-    def rate_meta_path(self, meta_path):
-        return 1
+    def _rate_meta_path(self, meta_path):
+        return 1.0
 
     def wants_to_continue(self):
-        return False
+        return self.number_of_calls < 5
 
 
 class ActiveLearner:
     meta_paths = None
     algorithm_type = None
     oracle = None
     batch_size = None
+    rating = None
 
-    def __init__(self, oracle=MockOracle(), algorithm=NO_RATING, batch_size=5):
-        self.meta_paths = self.fetch_meta_paths()
+    def __init__(self, oracle=MockOracle(), algorithm=NO_USER_FEEDBACK, batch_size=5, data_set=False):
+        self.meta_paths = data_set if data_set else self.fetch_meta_paths()
         self.oracle = oracle
         self.algorithm_type = algorithm
         self.batch_size = batch_size
+        self.rating = MetaPathRatingGraph()
 
     def retrieve_user_rating(self):
-        # TODO: How to compare???
-        if self.algorithm_type == NO_RATING:
-            # do let the user rate any meta_paths
-            return []
+        self._rate_paths()
+        self.rating.all_pair_distances()
+        return self.rating
+
+    def _rate_paths(self):
+        # consult oracle
+        if self.algorithm_type == NO_USER_FEEDBACK:
+            # do not let the user rate any meta_paths
+            pass
         if self.algorithm_type == BASELINE:
-            # take first batch
+            # produce rating of batches in random order, until user wants to stop
             interesting_meta_paths = self.meta_paths[:self.batch_size]
```

Comment:
Do we really want a random order or is a sensible order a `TODO`?

Reply:
I think the baseline is a random order and the `RESEARCH` mode is a `TODO`